ui/app/registration/controllers/patientCommonController.js

```diff
@@ -15,6 +15,63 @@ angular.module('bahmni.registration')
             $scope.genderCodes = Object.keys($rootScope.genderMap);
             $scope.dobMandatory = appService.getAppDescriptor().getConfigValue("dobMandatory") || false;
             $scope.readOnlyExtraIdentifiers = appService.getAppDescriptor().getConfigValue("readOnlyExtraIdentifiers");
+            $scope.showSaveConfirmDialogConfig = appService.getAppDescriptor().getConfigValue("showSaveConfirmDialog");
+
+            // Flag variable to check if Dont Save button is pressed or not
+            var flag = 0;
+
+            // Flag to check whether the button has an href or not
+            var hrefFlag = 0;
+
+            // Function is called when the Home Button is Clicked
+            $rootScope.homeNavigate = function () {
+                if ($scope.showSaveConfirmDialogConfig) {
+                    event.preventDefault();
+                    $scope.targetUrl = event.currentTarget.getAttribute('href');
+                    naviConfirmBox(event);
+                }
+            };
+
+            // Checks for State Change
+            var stateChangeListener = $rootScope.$on("$stateChangeStart", function (event, toState, toParams) {
+                naviConfirmBox(event, toState, toParams);
+            });
+
+            var naviConfirmBox = function (event, toState) {
+                if (flag === 0) {
+                    if ($scope.showSaveConfirmDialogConfig) {
+                        if (event) {
+                            event.preventDefault();
+                            if ($scope.targetUrl) {
+                                hrefFlag = 1;
+                            } else {
+                                $scope.targetUrl = toState.name;
+                                hrefFlag = 0;
+                            }
+                        }
+                        ngDialog.openConfirm({template: "views/navigationPrompt.html", scope: $scope});
+                    }
+                }
+            };
```

Comment:
@krishnanspace could you explain in as much detail as possible what the above does?

I don't like very much the fact that the controller refers to pure view-related concepts such as the existence of an `href`. While this can happen, it feels wrong. But please just explain what's behind the code and we'll take it from there.

Reply:
In the above code, the "ngDialog.openConfirm" opens the popup.The popup will contain the HTML defined by the template.The second parameter says that the popup's scope will be the same the scope of the current Controller

Reply:
Also shouldn't the HTML page be kept in a common directory?